omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleInstallResource1_8.java

```diff
@@ -0,0 +1,167 @@
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1 + "/moduleinstall", supportedClass = ModuleInstall.class, supportedOpenmrsVersions = {
+        "1.10.*", "1.11.*", "1.12.*", "1.8.*", "1.9.*", "2.0.*", "2.1.*" })
+public class ModuleInstallResource1_8 extends BaseDelegatingResource<ModuleInstall> implements Creatable {
+	
+	private ModuleFactoryWrapper moduleFactoryWrapper = new ModuleFactoryWrapper();
+	
+	private String moduleActionLink = ModuleActionResource1_8.class.getAnnotation(Resource.class).name();
+	
+	public void setModuleFactoryWrapper(ModuleFactoryWrapper moduleFactoryWrapper) {
+		this.moduleFactoryWrapper = moduleFactoryWrapper;
+	}
+	
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		moduleFactoryWrapper.checkPrivilege();
+		ModuleInstall moduleInstall = newDelegate();
+		setConvertedProperties(moduleInstall, post, getCreatableProperties(), true);
+		
+		String moduleUuid = moduleInstall.getModuleUuid();
+		String installUri = moduleInstall.getInstallUri();
+		Module module = null;
+		File moduleFile = null;
+		
+		ServletContext servletContext = getServletContext(context);
+		
+		if (moduleUuid == null && installUri == null) {
+			throw new IllegalRequestException("The moduleUuid and installUri is needed to perform this action");
+		}
+		
+		if (ResourceUtils.isUrl(installUri)) {
+			try {
+				Module existingModule = moduleFactoryWrapper.getModuleById(moduleUuid);
+				if (existingModule != null) {
+					List<Module> dependentModulesStopped = moduleFactoryWrapper.stopModuleAndGetDependent(existingModule);
+					
+					for (Module depMod : dependentModulesStopped) {
+						moduleFactoryWrapper.stopModuleSkipRefresh(depMod, servletContext);
+					}
+					
+					moduleFactoryWrapper.stopModuleSkipRefresh(existingModule, servletContext);
+					moduleFactoryWrapper.unloadModule(existingModule);
+				}
+				
+				URL downloadUrl = new URL(installUri);
+				String fileName = FilenameUtils.getName(downloadUrl.getPath());
+				InputStream inputStream = ModuleUtil.getURLStream(downloadUrl);
+				moduleFile = ModuleUtil.insertModuleFile(inputStream, fileName);
+				module = moduleFactoryWrapper.loadModule(moduleFile);
+				moduleFactoryWrapper.startModule(module, servletContext);
+				
+				return ConversionUtil.convertToRepresentation(moduleInstall, Representation.DEFAULT);
+			}
+			catch (MalformedURLException e) {
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+				throw new IllegalRequestException(e.getMessage());
```

Comment:
@malmike you should use a logger instead of `e.printStackTrace()` (like [here](https://github.com/openmrs/openmrs-module-webservices.rest/blob/639ad0273ee0c086cd823e3c118e657ec4f0b09e/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java#L558) for example).